compiler/rustc_passes/src/lang_items.rs

```diff
@@ -321,6 +270,11 @@ fn get_lang_items(tcx: TyCtxt<'_>, (): ()) -> LanguageItems {
     // Find all required but not-yet-defined lang items.
     weak_lang_items::check_crate(tcx, &mut items);
 
+    if collector.has_lang_item_generic_error {
+        // Continuing compilation with a malformed lang item will not end well, so stop as soon as possible
+        tcx.sess.abort_if_errors();
```

Comment:
Compilation should probably be aborted if we have the wrong target also.

Reply:
That's handled by not adding the lang item to the list in the first place, so it can never be used incorrectly in the compiler.  But that actually gives me an idea to handle this better than a flag in the struct, I can probably remove these incorrect generics too, and that will allow compilation to continue a little and potentially show other errors, which is nicer.